.github/workflows/check-prdoc.yml

```diff
@@ -22,49 +22,33 @@ jobs:
   check-prdoc:
     runs-on: ubuntu-latest
     timeout-minutes: 10
-    if: github.event.pull_request.number != ''
     steps:
       - name: Checkout repo
         uses: actions/checkout@6d193bf28034eafb982f37bd894289fe649468fc #v4.1.7
-      # we cannot show the version in this step (ie before checking out the repo)
-      # due to https://github.com/paritytech/prdoc/issues/15
-      - name: Check if PRdoc is required
-        id: get-labels
+      - name: Check prdoc format
         run: |
           echo "Pulling $IMAGE"
           $ENGINE pull $IMAGE
 
-          # Fetch the labels for the PR under test
-          echo "Fetch the labels for $API_BASE/${REPO}/pulls/${GITHUB_PR}"
-          labels=$( curl -H "Authorization: token ${GITHUB_TOKEN}" -s "$API_BASE/${REPO}/pulls/${GITHUB_PR}" | jq '.labels | .[] | .name' | tr "\n" ",")
-          echo "Labels: ${labels}"
-          echo "labels=${labels}" >> "$GITHUB_OUTPUT"
-
           echo "Checking PRdoc version"
           $ENGINE run --rm -v $PWD:/repo $IMAGE --version
 
-      - name: Early exit if PR is silent
-        if: ${{ contains(steps.get-labels.outputs.labels, 'R0') }}
-        run: |
-          hits=$(find prdoc -name "pr_$GITHUB_PR*.prdoc" | wc -l)
-          if (( hits > 0 )); then
-            echo "PR detected as silent, but a PRDoc was found, checking it as information"
-            $ENGINE run --rm -v $PWD:/repo $IMAGE check -n ${GITHUB_PR} || echo "Ignoring failure"
-          else
-            echo "PR detected as silent, no PRDoc found, exiting..."
-          fi
-          echo "If you want to add a PRDoc, please refer to $PRDOC_DOC"
-          exit 0
+          echo "Check prdoc format"
+          echo "For PRDoc format, please refer to $PRDOC_DOC"
+          $ENGINE run --rm -v $PWD:/repo -e RUST_LOG=info $IMAGE check
 
-      - name: PRdoc check for PR#${{ github.event.pull_request.number }}
-        if: ${{ !contains(steps.get-labels.outputs.labels, 'R0') }}
+      - name: Check if PRdoc is required
+        if: github.event.pull_request.number != ''
+        id: get-labels
         run: |
-          echo "Checking for PR#${GITHUB_PR}"
-          echo "You can find more information about PRDoc at $PRDOC_DOC"
-          $ENGINE run --rm -v $PWD:/repo -e RUST_LOG=info $IMAGE check -n ${GITHUB_PR}
+          # Fetch the labels for the PR under test
+          echo "Fetch the labels for $API_BASE/${REPO}/pulls/${GITHUB_PR}"
+          labels=$( curl -H "Authorization: token ${GITHUB_TOKEN}" -s "$API_BASE/${REPO}/pulls/${GITHUB_PR}" | jq '.labels | .[] | .name' | tr "\n" ",")
+          echo "Labels: ${labels}"
+          echo "labels=${labels}" >> "$GITHUB_OUTPUT"
 
       - name: Validate prdoc for PR#${{ github.event.pull_request.number }}
-        if: ${{ !contains(steps.get-labels.outputs.labels, 'R0') }}
+        if: ${{ github.event.pull_request.number != '' && !contains(steps.get-labels.outputs.labels, 'R0') }}
```

Comment:
There could be `R0` with PR doc. Maybe we should also validate it?

Reply:
Yes true, but at the same time R0 means it doesn't show in the changelog so change should be minor.